app/models/package_manager/maven.rb

```diff
@@ -230,20 +232,20 @@ def self.licenses(xml)
         .map(&:last)
     end
 
-    def self.latest_version(versions, name)
-      if versions.present?
-        versions
-          .max_by { |version| version[:published_at] }
-          .dig(:number)
-      else
-        # TODO: this is in place to handle packages that are no longer on maven-repository.com
-        # this could be removed if we switched to a package data provider that supplied full information
-        Project
-          .find_by(name: name, platform: formatted_name)
-          &.versions
-          &.max_by(&:published_at)
-          &.number
-      end
+    def self.latest_version(name)
+      maven_metadata = get_raw(MavenUrl.from_name(name, repository_base, NAME_DELIMITER).maven_metadata)
+      latest = Nokogiri::XML(maven_metadata)
+        .css("versioning > latest, versioning > release, metadata > version")
+        .map(&:text)
+        .first
+
+      # TODO: this is in place to handle packages that are no longer on maven-repository.com
```

Comment:
We're not using maven-repository now right? Maybe just delete this, and if it fails we can explore. Also if we need it, that query below can be a single pluck instead of hydrating hundreds/thousands of version objects to grab a single version number.

Reply:
`Version.where(project: Project.where(name: project.name, platform: project.platform)).order(published_at: :desc).limit(1).pluck(:number)`